File: src/1662_check_if_two_string_arrays_are_equivalent/CheckIfTwoStringArraysAreEquivalent.java
public class CheckIfTwoStringArraysAreEquivalent {
    public boolean arrayStringsAreEqual(String[] word1, String[] word2) {
        return String.join("", word1).equals(String.join("", word2));
    }

    public static void main(String[] args) {
        CheckIfTwoStringArraysAreEquivalent sol = new CheckIfTwoStringArraysAreEquivalent();
        System.out
                .println(sol.arrayStringsAreEqual(new String[] { "ab", "c" }, new String[] { "a", "bc" }) + " | True");
        System.out
                .println(sol.arrayStringsAreEqual(new String[] { "a", "cb" }, new String[] { "ab", "c" }) + " | False");
        System.out.println(
                sol.arrayStringsAreEqual(new String[] { "abc", "d", "defg" }, new String[] { "abcddefg" }) + " | True");
        System.out.println(sol.arrayStringsAreEqual(new String[] { "" }, new String[] { "" }) + " | True");
        System.out
                .println(sol.arrayStringsAreEqual(new String[] { "a", "b", "c" }, new String[] { "abc" }) + " | True");
    }
}

File: src/1662_check_if_two_string_arrays_are_equivalent/check_if_two_string_arrays_are_equivalent.py
from typing import List

class Solution:
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
        return ''.join(word1) == ''.join(word2)

def test():
    sol = Solution()
    print(sol.arrayStringsAreEqual(["ab", "c"], ["a", "bc"]), "| True")
    print(sol.arrayStringsAreEqual(["a", "cb"], ["ab", "c"]), "| False")
    print(sol.arrayStringsAreEqual(["abc", "d", "defg"], ["abcddefg"]), "| True")
    print(sol.arrayStringsAreEqual([""], [""]), "| True")
    print(sol.arrayStringsAreEqual(["a", "b", "c"], ["abc"]), "| True")

if __name__ == "__main__":
    test()
